src/System.Drawing.Primitives/src/System/Drawing/Color.cs

```diff
@@ -420,188 +420,119 @@ private long Value
             }
         }
 
+        private static void ThrowOutOfByteRange(int value, string name) =>
+            throw new ArgumentException(SR.Format(SR.InvalidEx2BoundArgument, name, value, byte.MinValue, byte.MaxValue));
+
         private static void CheckByte(int value, string name)
         {
-            if (value < 0 || value > 255)
-                throw new ArgumentException(SR.Format(SR.InvalidEx2BoundArgument, name, value, 0, 255));
+            if ((uint)value > byte.MaxValue)
+                ThrowOutOfByteRange(value, name);
         }
 
-        private static long MakeArgb(byte alpha, byte red, byte green, byte blue) =>
-            (long)unchecked((uint)(red << ARGBRedShift |
+        private static uint MakeArgb(byte alpha, byte red, byte green, byte blue) =>
+            unchecked((uint)(red << ARGBRedShift |
                 green << ARGBGreenShift |
                 blue << ARGBBlueShift |
-                alpha << ARGBAlphaShift)) & 0xffffffff;
+                alpha << ARGBAlphaShift));
 
-        public static Color FromArgb(int argb) => new Color(argb & 0xffffffff, StateARGBValueValid, null, 0);
+        private static Color FromArgb(uint argb) => new Color(argb, StateARGBValueValid, null, (KnownColor)0);
+
+        public static Color FromArgb(int argb) => FromArgb((uint)argb);
```

Comment:
What is the perf benefit of adding and extra call to `FromArgb(uint)`? we could just change the current call to `new Color((uint)argb, StateARGBValueValid, null, (KnownColor)0);`